feat: add TypeScript support for hooks and filters (#519)
asyncapi/generator: docs/authoring.md

```

In case you have more than one template and want to reuse filters, you can put them in a single library. You can configure such a library in the template configuration under `filters` property. You can also use the official AsyncAPI [filters library](https://github.com/asyncapi/generator-filters). To learn how to add such filters to configuration [read more about the configuration file](#configuration-file).

## TypeScript support 

The AsyncAPI generator has TypeScript support for [hooks](#hooks) and Nunjucks's [filters](#filters). Assumptions:

- Installing the `typescript` package and creating the` tsconfig.json` file isn't necessary.
- Source code of the hook/filter must have `.ts` extension.

> - Each package related with the typings for TypeScript like `@types/node` must be installed in the template under `dependencies` array. This is because the Generator transpiles the TypeScript code on-the-fly while rendering the template, so cannot uses packages under `devDependencies`.

Reviewer: ```suggestion
- Each package related to the typings for TypeScript like `@types/node` must be installed in the template under `dependencies` array. This is because the Generator transpiles the TypeScript code on-the-fly while rendering the template, and cannot use packages under `devDependencies`.
```

> - Each template should have installed `@types/node` package to support typings for Node.

Reviewer: ```suggestion
- Each template should have `@types/node` package installed to enable support for typings for Node.
```

> - Each package should have installed `@types/node` package to support typings for Node.

Reviewer: what package?

Author: Sorry, here should be `template` -> `Each template...`
